### PR TITLE
DOC: orthogonal_procrustes fix date of reference paper and DOI

### PR DESCRIPTION
**Pull request by arch-dawson**
Changed reference date from 1996 to 1966, see link below for confirmation.

https://link.springer.com/article/10.1007/BF02289451

#### Reference issue
Did not raise an issue

#### What does this implement/fix?
Fixes date in a reference

#### Additional information
N/A

### CODE REVIEW

**Comment by ilayn** [comment]
Thank you @arch-dawson. While we are at it, could you please add the DOI while respecting the 80 character length (possibly a comma and a linebreak would do) as instructed here https://docs.scipy.org/doc/scipy/dev/contributor/rendering_documentation.html#dois-in-references ? 

So it would be ``:doi:`10.1007/BF02289451` ``

**Review by tupui — APPROVED** [review]
Thank you @arch-dawson, good catch 😃

File: scipy/linalg/_procrustes.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def orthogonal_procrustes(A, B, check_finite=True):
     References
     ----------
     .. [1] Peter H. Schonemann, "A generalized solution of the orthogonal
-           Procrustes problem", Psychometrica -- Vol. 31, No. 1, March, 1996.
+           Procrustes problem", Psychometrica -- Vol. 31, No. 1, March, 1966.
```
Comment (tupui) [inline scipy/linalg/_procrustes.py:56]:
```suggestion
           Procrustes problem", Psychometrica -- Vol. 31, No. 1, March, 1966.
           :doi:`10.1007/BF02289451`
```

**Comment by tupui** [comment]
Just saw your suggestion @ilayn and took the liberty to just do it quick.

**Comment by ilayn** [comment]
Thanks all

**Pull request merged**